src/loan/LendingTerm.sol

```diff
@@ -0,0 +1,367 @@
+        require(borrowAmount <= maxBorrowLtv, "LendingTerm: not enough LTV buffer");
+
+        // check that this lending term is active
+        address _guildToken = guildToken;
+        require(GuildToken(_guildToken).isGauge(address(this)), "LendingTerm: terms unavailable");
+
+        // check the debt ceiling & hardcap
+        uint256 _totalBorrows = totalBorrows();
+        require(_totalBorrows + borrowAmount <= hardCap, "LendingTerm: hardcap reached");
+        uint256 _totalSupply = ERC20(creditToken).totalSupply();
+        if (_totalSupply != 0) {
+            uint256 debtCeiling = GuildToken(_guildToken).calculateGaugeAllocation(address(this), _totalSupply + borrowAmount);
+            require(_totalBorrows + borrowAmount <= debtCeiling, "LendingTerm: debt ceiling reached");
+        }
+
+        // save loan in state
+        loans[loanId] = Loan({
+            borrower: msg.sender,
+            borrowAmount: borrowAmount,
+            collateralAmount: collateralAmount,
+            caller: address(0),
+            callTime: 0,
+            originationTime: block.timestamp,
+            closeTime: 0
+        });
+
+        // mint CREDIT to the borrower
+        RateLimitedCreditMinter(creditMinter).mint(msg.sender, borrowAmount);
+
+        // update total borrows
+        totalBorrowsStored = _totalBorrows + borrowAmount;
+        totalBorrowsLastUpdate = block.timestamp;
+
+        // pull the collateral from the borrower
+        ERC20(collateralToken).transferFrom(msg.sender, address(this), collateralAmount);
+    }
+
+    /// @notice repay an open loan
+    function repay(bytes32 loanId) external {
+        Loan storage loan = loans[loanId];
+
+        // check the loan is open
+        uint256 originationTime = loan.originationTime;
+        require(originationTime != 0, "LendingTerm: loan not found");
+        require(originationTime < block.timestamp, "LendingTerm: loan opened in same block");
+        require(loan.closeTime == 0, "LendingTerm: loan closed");
+
+        // compute interest owed
+        uint256 loanDebt = getLoanDebt(loanId);
+    
+        // if the loan is called and we are within the call period, deduce the callFee from
+        // the amount of debt to repay
+        uint256 callTime = loan.callTime;
+        if (callTime != 0 && block.timestamp <= callTime + callPeriod) {
+            loanDebt -= getLoanCallFee(loanId);
+        }
+        
+        // pull the total CREDIT owed and refill the buffer of available CREDIT mints
+        address _creditToken = creditToken;
+        ERC20(_creditToken).transferFrom(msg.sender, address(this), loanDebt);
+        RateLimitedCreditMinter(creditMinter).replenishBuffer(loanDebt);
+        CreditToken(_creditToken).burn(loanDebt);
+
+        // close the loan
+        loan.closeTime = block.timestamp;
+
+        // update total borrows
+        totalBorrowsStored = totalBorrows() - loanDebt;
+        totalBorrowsLastUpdate = block.timestamp;

```

Comment:
what happens here if the loan has been called ? loanDebt is discounted by the callFee so totalBorrowsStored might not be correct